packages/wallet-sdk/src/sign/scw/utils/createSubAccountSigner.ts

```diff
@@ -42,22 +43,22 @@ export async function createSubAccountSigner(subaccount: SubAccountInfo) {
     ownerIndex: index,
     address: subaccount.address,
     client,
-    factoryData: subaccount.initCode.factoryCalldata,
+    factoryData: subaccount.factoryData,
   });
 
   return {
     request: async (args: RequestArguments): Promise<Hex | Address[] | number | SubAccountInfo> => {
       switch (args.method) {
-        case 'wallet_addAddress':
+        case 'wallet_addSubAccount':
           return subaccount;
         case 'eth_accounts':
           return [subaccount.address] as Address[];
         case 'eth_coinbase':
           return subaccount.address;
         case 'net_version':
-          return subaccount.chainId;
+          return numberToHex(chainId);
```

Comment:
https://docs.metamask.io/services/reference/ethereum/json-rpc-methods/net_version/ 

think this should be decimal number string